### DISCUSSION TITLE
F' and Dynamic Priority

### DISCUSSION THREAD

**Discussion by rawinza555** [nasa/fprime#1100] [General]
Does F' support dynamic priority for the thread? In other words, is there a way for changing the priority of each component's thread based on certain condition? For example, I might want to have anything related to comm to have higher priority if the satellite is closed by the ground station coverage area and then set it back to low priority afterward.

**Comment by timcanham** [comment]
There is nothing in the API to change the priority dynamically. Consider that your comm task will more likely be idle when you are not doing comms, so it won't consume as many CPU cycles. If you leave it at a higher priority all the time, it will ramp up naturally when you are in range.

**Reply by rawinza555** [reply]
Thanks!
